src/sentry/deletions/defaults/group.py

```diff
@@ -47,8 +47,9 @@ def delete_instance(self, instance):
         return super(GroupDeletionTask, self).delete_instance(instance)
 
     def mark_deletion_in_progress(self, instance_list):
-        from sentry.models import GroupStatus
+        from sentry.models import Group, GroupStatus
 
-        for instance in instance_list:
-            if instance.status != GroupStatus.DELETION_IN_PROGRESS:
-                instance.update(status=GroupStatus.DELETION_IN_PROGRESS)
+        in_progress = Group.objects.filter(
```

Comment:
As a nit, there's no reason to have this extra variable. This can just be one statement.

```python
Group.objects.filter(
    id__in=[i.id for i in instance_list],
).exclude(
    status=GroupStatus.DELETION_IN_PROGRESS,
).update(
    status=GroupStatus.DELETION_IN_PROGRESS,
)
```

Reply:
What @tkaemming said too.

Reply:
😎